src/main/java/org/folio/service/InterfaceService.java

```diff
@@ -1,79 +1,76 @@
 package org.folio.service;
 
 import static javax.ws.rs.core.Response.Status.NOT_FOUND;
-import static org.folio.util.ResponseUtils.handleFailure;
-import static org.folio.util.ResponseUtils.handleNoContentResponse;
 
 import javax.ws.rs.core.Response;
 
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.folio.persist.CriterionBuilder;
-import org.folio.persist.Tx;
+import org.folio.rest.persist.Conn;
 import org.folio.rest.persist.PostgresClient;
 import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.util.ResponseUtils;
 
 import io.vertx.core.AsyncResult;
-import io.vertx.core.Context;
 import io.vertx.core.Future;
 import io.vertx.core.Handler;
 import io.vertx.core.Promise;
 import io.vertx.ext.web.handler.HttpException;
 
 public class InterfaceService {
 
+  private static final Logger logger = LogManager.getLogger(InterfaceService.class);
+
   private static final String INTERFACE_CREDENTIALS_TABLE = "interface_credentials";
   private static final String INTERFACE_TABLE = "interfaces";
 
-  private final Logger logger = LogManager.getLogger(this.getClass());
   private final PostgresClient pgClient;
 
   public InterfaceService(PostgresClient pgClient) {
     this.pgClient = pgClient;
   }
 
-  public void deleteOrganizationsInterfaceById(String id,
-    Context vertxContext, Handler<AsyncResult<Response>> asyncResultHandler) {
-    vertxContext.runOnContext(v -> {
-      Tx<String> tx = new Tx<>(id, pgClient);
-      tx.startTx()
-        .compose(this::deleteCredentialByInterfaceId)
-        .compose(this::deleteInterfaceById)
-        .compose(Tx::endTx)
-        .onComplete(handleNoContentResponse(asyncResultHandler, tx, "Interface '{}' '{}' deleted"));
-    });
+  public void deleteOrganizationsInterfaceById(String id, Handler<AsyncResult<Response>> asyncResultHandler) {
+    pgClient.withTrans(
+      conn ->
+        deleteCredentialByInterfaceId(conn, id)
+          .compose(conn1 -> deleteInterfaceById(conn1, id))
+          .onComplete(result -> ResponseUtils.handleNoContentResponse(result, id, "Interface '{}' '{}' deleted", asyncResultHandler))
+    );
```

Comment:
.onComplete should be invoked after transaction committed or reverted, so ')' should be moved after the last compose.

Also minor thing, as I know we use conn on the same line with pgClient.withTrans() so I propose to follow the same style

Reply:
the style we are using, is a little bit confusing and give impression .onCopmlete in scope of withTrans() operation. so need to think new style for me these styles is more clear to understand 
```
 PgUtil.postgresClient(vertxContext, okapiHeaders)
        .withTrans(conn -> conn.execute(createDeleteQuery(wrapper, okapiHeaders))
          .compose(rows -> {
            if (rows.rowCount() != 0) {
              List<User> users = new ArrayList<>();
              rows.iterator().forEachRemaining(row -> users.add(new User().withId(row.getUUID(USER_ID).toString())));
              return userOutboxService.saveUserOutboxLogForDeleteUsers(conn, users, okapiHeaders);
            }
            return Future.succeededFuture();
          }))
        .map(bVoid -> DeleteUsersByUserIdResponse.respond204())
        .map(Response.class::cast)
        .onComplete(reply -> {
          userOutboxService.processOutboxEventLogs(vertxContext.owner(), okapiHeaders);
          asyncResultHandler.handle(reply);
        });
```
```
    pgClient.withTrans(conn -> usersService.getUserByIdForUpdate(conn, entity.getId())
      .compose(userFromStorage -> {
        if (userFromStorage == null) {
          return succeededFuture(PutUsersByUserIdResponse.respond404WithTextPlain(entity.getId()));
        }

        return userTenantService.validateUserAcrossTenants(entity, userFromStorage, okapiHeaders, conn, vertxContext)
          .compose(aVoid -> usersService.updateUser(conn, entity)
            .compose(user -> userOutboxService.saveUserOutboxLogForUpdateUser(conn, user, userFromStorage, okapiHeaders))
            .map(isUserOutboxLogSaved -> PutUsersByUserIdResponse.respond204())
            .map(Response.class::cast));
      })
    ).onComplete(reply -> {
      if (reply.cause() != null) {
        handleUpdateUserFailures(entity, asyncResultHandler, reply);
        return;
      }
      userOutboxService.processOutboxEventLogs(vertxContext.owner(), okapiHeaders);
      asyncResultHandler.handle(reply);
    });
  }
```

Reply:
Agree, in particular `compose` for me has one more tab and comparing to `onComplete` which is not the case in other places.